modules/3d/src/octree.cpp

```diff
@@ -175,11 +175,16 @@ bool Octree::create(const std::vector<Point3f> &pointCloud, int _maxDepth)
     this->p->size = 2 * halfSize;
 
     // Insert every point in PointCloud data.
+    int cnt=0;
     for(size_t idx = 0; idx < pointCloud.size(); idx++ )
     {
-        insertPoint(pointCloud[idx]);
+        if(!insertPoint(pointCloud[idx])){
+            cnt++;
+        };
+    }
+    if(cnt!=0){
+        CV_LOG_WARNING(NULL,"OverAll "<<cnt<<" points has been ignored!");
```

Comment:
There is `CV_LOG_IF_WARNING` macro for this purpose (`CV_LOG_*` is a whole body of `if` statement).

Also it makes sense to show `pointCloud.size()` too.

Reply:
Thanks, I quite agree with you.